component/class/defaults.yml

```
@@ -1,10 +1,14 @@
 parameters:
   exoscale_metrics_collector:
+    =_metadata:
+      multi_instance: true
     secrets:
-      exoscale:
+      cronjob:
         stringData:
           api_key: "?{vaultkv:${cluster:tenant}/${cluster:name}/exoscale-metrics-collector/key}"
           api_secret: "?{vaultkv:${cluster:tenant}/${cluster:name}/exoscale-metrics-collector/secret}"
+          cluster_server: 'https://api.exoscale-ch-gva-2-0.appuio.cloud:6443'
+          cluster_token: "?{vaultkv:${cluster:tenant}/c-appuio-exoscale-ch-gva-2-0/exoscale-metrics-collector/cluster-token}"
```

I would suggest to directly use the environment variables as required by the job.
Also, I wouldn't put a specific cluster in the defaults here. Rather, I'd add a tutorial/how-to page where it's described how to fill in the URL and token (with vault reference in K8S_TOKEN)
```suggestion
          EXOSCALE_API_KEY: "?{vaultkv:${cluster:tenant}/${cluster:name}/exoscale-metrics-collector/key}"
          EXOSCALE_API_SECRET: "?{vaultkv:${cluster:tenant}/${cluster:name}/exoscale-metrics-collector/secret}"
          K8S_SERVER_URL: ''
          K8S_TOKEN: ""
```

---

I would argue that the same should be done for the exoscale credentials.

---

Hm, I had an idea again:
```suggestion
          EXOSCALE_API_KEY: "?{vaultkv:${cluster:tenant}/${cluster:name}/exoscale-metrics-collector/key}"
          EXOSCALE_API_SECRET: "?{vaultkv:${cluster:tenant}/${cluster:name}/exoscale-metrics-collector/secret}"
          K8S_SERVER_URL: ''"
          K8S_TOKEN: "?{vaultkv:${cluster:tenant}/${cluster:name}/exoscale-metrics-collector/${_instance}/cluster-token}"
```
So that way the K8s token is also parameterized.
Still leaves the server URL, but I don't think it makes much sense to put this in Vault. But if some commands are in a copy-pastable how-to, then it'd be fine with me.

---

I would put the k8s server in vault as well at this point and would also create a how to page. Also `${_instance}` for exoscale credentials

---

Fine by me. In that case the how-to page is missing in this PR, or in a follow-up

---

I am working on that now and will be added here.